17-lifecycle-lab/src/01-cps.jsx

```diff
@@ -7,12 +7,60 @@
  */
 
 import React from 'react';
+import Indicator from './02-cps-with-indicator'
+export default class Cps  extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            seconds: props.initialSeconds || 0,
+            numberOfClicks: props.initialNumberOfClicks || 0,
+        };
+    }
+
+    tick = () => {
+        this.setState({seconds: this.state.seconds + 1});
+    }
+
+    componentWillMount() {
+        this.timer = setInterval(this.tick, 1000);
+
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    setSeconds(seconds) {
+        this.setState({seconds});
+    }
+
+    increaseClickCount = () => {
+        this.setState({numberOfClicks: this.state.numberOfClicks + 1});
+    }
+
+    getColor = () =>
+    {
+        if ((this.state.numberOfClicks/this.state.seconds)>1) {
+            return "red";
+        }
+        else
+        {
+            return "green";
+        }
+    }
 
-export default createClass({
   render() {
+        const cpsRate=this.state.numberOfClicks/this.state.seconds;
+        const tooFast=cpsRate>1;
+        const color =this.getColor();
     return (<div>
-      <button>Click Fast</button>
-      <p>CPS rate: {0}</p>
+      <button onClick={this.increaseClickCount}>Click Fast</button>
+      <p>CPS rate: {cpsRate}</p>
+      <p> {tooFast && "לא כל כך מהר..."} {!tooFast && "יותר מהר!"}</p>
+        <Indicator color={color}/>
```

Comment:
It's better to pass indicator predicate (boolean) and let it define it's styling inside the indicator based on predicate. Indicator styling is not a concern of parent component.